capsules/src/net/ipv6/ipv6.rs

```diff
@@ -343,11 +346,18 @@ impl IPPayload<'a> {
     /// `(u8, u16)` - Returns a tuple of the `ip6_nh` type of the
     /// `transport_header` and the total length of the `IPPayload`
     /// (when serialized)
-    pub fn set_payload(&mut self, transport_header: TransportHeader, payload: &[u8]) -> (u8, u16) {
+    pub fn set_payload(
+        &mut self,
+        transport_header: TransportHeader,
+        payload: &LeasableBuffer<'static, u8>,
+    ) -> (u8, u16) {
         if self.payload.len() < payload.len() {
-            // TODO: Error
+            debug!("Error in set_payload for ipv6_packet");
```

Comment:
If I understand correctly, two lines later this is going to `panic!` when `self.payload[i]` goes beyond valid array bounds -- maybe this should return here? Or is the goal to leave the `panic!` in for now until better error-handling bubbles through the rest of the networking stack?

Reply:
Goal for now is to leave panic!(), the whole networking stack definitely needs improved error handling eventually but don't want to worry about that in this PR.